terraform/trial_rg/main.tf

```diff
@@ -20,65 +20,95 @@ resource "azurerm_app_service_plan" "apps_service_plan" {
     tier = "Standard"
     size = "S1"
   }
+
+  depends_on = [
+    azurerm_resource_group.trial_rg,
+  ]
 }
 
-# Site service
-module "trial_app_service_site" {
-  source              = "./modules/genericservice"
-  app_name            = "site"
-  rg_name             = azurerm_resource_group.trial_rg.name
-  app_service_plan_id = azurerm_app_service_plan.apps_service_plan.id
-  trial_name          = var.trial_name
-  environment         = var.environment
-  docker_image        = var.site_image_name
-  docker_image_tag    = var.site_image_tag
+# Audit storage
+module "trial_audit" {
+  source     = "./modules/audit"
+  trial_name = var.trial_name
+  rg_name    = azurerm_resource_group.trial_rg.name
+
+  depends_on = [
+    azurerm_resource_group.trial_rg,
+  ]
 }
 
-# Practitioner service
-module "trial_app_service_practitioner" {
-  source              = "./modules/genericservice"
-  app_name            = "practitioner"
+# Fhir server
+module "fhir_server" {
+  source              = "./modules/fhir"
+  trial_name          = var.trial_name
   rg_name             = azurerm_resource_group.trial_rg.name
   app_service_plan_id = azurerm_app_service_plan.apps_service_plan.id
-  trial_name          = var.trial_name
-  environment         = var.environment
-  docker_image        = var.practitioner_image_name
-  docker_image_tag    = var.practitioner_image_tag
+
+  depends_on = [
+    azurerm_app_service_plan.apps_service_plan,
+  ]
+}
+
+# vnet
+module "trial_vnet" {
+  source      = "./modules/vnet"
+  trial_name  = var.trial_name
+  rg_name     = azurerm_resource_group.trial_rg.name
+
+  depends_on = [
+    azurerm_resource_group.trial_rg,
+  ]
 }
 
 # Key vault
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault
 module "trial_keyvault" {
   source      = "./modules/kv"
   trial_name  = var.trial_name
   environment = var.environment
   rg_name     = azurerm_resource_group.trial_rg.name
   tenant_id   = "99804659-431f-48fa-84c1-65c9609de05b"
+  subnet_id   = module.trial_vnet.subnet_id
+
+  depends_on = [
+    module.trial_vnet,
+  ]
 }
 
-# Audit storage
-module "trial_audit" {
-  source     = "./modules/audit"
-  trial_name = var.trial_name
-  rg_name    = azurerm_resource_group.trial_rg.name
+ # Add vnet integration to sql databases
+resource "azurerm_sql_virtual_network_rule" "sqlvnetrule" {
```

Comment:
this seems to correspond to service endpoint and not a private endpoint which is what i think we want here.

Reply:
added private endpoint and private link.